### PR TITLE
`crackle dev`: add a `--shim` option

### PR DESCRIPTION
**Pull request by mrm007**
Add a `crackle dev --shim` option with choices `none` and `require` (default).
The `crackle dev --webpack` option has been removed, replaced with `crackle dev --shim=none`.

This is needed in environments where a bundler is used (tested with Webpack and Vite). The bundler doesn't need the [`tsx` loader](https://github.com/privatenumber/tsx?tab=readme-ov-file#nodejs-loader) because it can process TypeScript files directly. 

Previously, we were generating ESM-incompatible code which only worked in Webpack [using a hack](https://github.com/SEEK-Jobs/metropolis/blob/f3133389444267e7da4a6b4fb8329b9da22d76ea/playroom.config.js#L26-L30).

```diff
// dist/index.mjs

-const _mod = require("../src/index.ts");
+import * as _mod from "../src/index.ts";

 export const something = _mod.something;
 export default _mod.default;
```

### CODE REVIEW

File: .changeset/cli-dev-shim.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@crackle/cli': minor
+---
+
+Add a `crackle dev --shim` option with choices `none` and `require` (default)
+
+The `crackle dev --webpack` option has been removed, replaced with `crackle dev --shim=none`
```
Comment (michaeltaranto) [inline .changeset/cli-dev-shim.md:7]: Given this is a breaking change should maybe make this more visible. I understand we are pre 1.0, but maybe something like:
````suggestion
### BREAKING CHANGE:
The `--webpack` option has been removed from the `dev` command in favour of the more generic `--shim` option. 

Consumers should migrate to use `--shim=none` as follows:
```diff
- crackle dev --webpack
+ crackle dev --shim=none
```
````

File: packages/core/src/config.ts [hunk 2]
```diff
@@ -56,11 +56,14 @@ export interface Config {
...
+     * Controls whether Crackle should generate a shim.
      *
-     * @default false
+     * - 'require' creates a `require` shim using `createRequire`
+     * - 'none' does not add a shim
+     *
+     * @default 'require'
      */
-    webpack?: boolean;
+    shim?: 'require' | 'none';
```
Comment (michaeltaranto) [inline packages/core/src/config.ts:66]: Worth adding to the readme?

**Review by michaeltaranto — APPROVED** [review]
Just some docs and changeset comments

File: .changeset/cli-dev-shim.md (hunk 1 above)
Comment (mrm007) [reply .changeset/cli-dev-shim.md:7]: We're still on `0.x`, so I don't think it's worth drawing attention to the breaking change. I will update the wording 👍

**Comment by mrm007** [comment]
I'll update the README for this and https://github.com/seek-oss/crackle/pull/167 in a follow-up PR

**Pull request merged**